functional_tests/steps/bundles.py

```diff
@@ -53,3 +54,30 @@ def the_user_opens_page_chooser(context: Context) -> None:
 def the_locale_column_is_displayed(context: Context) -> None:
     modal = context.page.locator(".modal-body")
     modal.get_by_role("columnheader", name="Locale").is_visible()
+
+
+@step("the user can add Bundles created by user")
+def a_bundle_with_name_item_name_has_been_created_by_username(context: Context) -> None:
+    context.bundle = BundleFactory(created_by=context.user_data["user"])
```

Comment:
This step sounds like an assertion (the user can do something), so it should use `@then` rather than `@step`. However you are not making an assertion in the code, you are creating a bundle.

If this is a pre-condition, then this should use `@given` and the step should be reworded (e.g. "a user created a bundle").

Your functional tests should roughly follow this sequence:

**Given** _pre-conditions_ (data modification allowed here to setup the scenario) using `@given` and occasionally `@when`
**When** _user actions_ (data modification not allowed here, only user actions) using `@when` and `@step`
**Then** _assertions_ (we check the presence or lack-of different things) using `@then`